pkg/api/customization/cluster/actions_cis.go

```diff
@@ -73,16 +73,51 @@ func (a ActionHandler) runCisScan(actionName string, action *types.Action, apiCo
 			fmt.Sprintf("CIS scan already running on cluster"))
 	}
 
-	if cisScanConfig.OverrideBenchmarkVersion != "" {
-		_, err := a.CisBenchmarkVersionLister.Get(namespace.GlobalNamespace, cisScanConfig.OverrideBenchmarkVersion)
-		if err != nil {
-			if kerrors.IsNotFound(err) {
+	validBenchmarkVersions, err := a.getValidBenchmarkVersions()
+	if err != nil {
+		logrus.Errorf("error fetching benchmark versions: %v", err)
+		return httperror.WrapAPIError(err, httperror.ServerError, "error fetching cis benchmark versions %v")
+	}
+
+	if cisScanConfig.OverrideBenchmarkVersion != "" &&
+		!validBenchmarkVersions[cisScanConfig.OverrideBenchmarkVersion] {
+		return httperror.WrapAPIError(err, httperror.InvalidOption,
+			fmt.Sprintf("invalid override benchmark version specified"))
+	}
+
+	userContext, err := a.ClusterManager.UserContext(apiContext.ID)
+	if err != nil {
+		logrus.Errorf("error fetching user context for cluster %v: %v", apiContext.ID, err)
+		return httperror.WrapAPIError(err, httperror.ServerError,
+			fmt.Sprintf("error fetching user contect for cluster %v", apiContext.ID))
+	}
+
+	cm, err := userContext.Core.ConfigMaps(v3.DefaultNamespaceForCis).Controller().Lister().Get(v3.DefaultNamespaceForCis, v3.ConfigMapNameForUserConfig)
+	if err != nil && !kerrors.IsNotFound(err) {
+		logrus.Errorf("error fetching configmap %v: %v", v3.ConfigMapNameForUserConfig, err)
+		return httperror.WrapAPIError(err, httperror.ServerError,
+			fmt.Sprintf("error fetching configmap %v", v3.ConfigMapNameForUserConfig))
+	}
+	// check for config.json
+	skipInfoStr, ok := cm.Data[DefaultConfigFilename]
+	if ok {
+		// check for content
+		skipInfo := map[string]map[string][]string{}
+		if err := json.Unmarshal([]byte(skipInfoStr), &skipInfo); err != nil {
+			logrus.Errorf("error unmarshaling cis skip info from configmap: %v", err)
+			return httperror.WrapAPIError(err, httperror.InvalidFormat, fmt.Sprintf("invalid skip info found in configmap: %v", v3.ConfigMapNameForUserConfig))
+		}
+		versionsMap, ok := skipInfo["skip"]
+		if !ok {
+			return httperror.WrapAPIError(err, httperror.InvalidOption,
```

Comment:
Do we want to wrap err here?

Reply:
copy, paste error, thanks for the catch.